js/components/root.js

```diff
@@ -7,6 +7,8 @@
  * later. See the COPYING file.
  */
 
+require('./notification');
```

Comment:
Actually, the intended use is that the require call returns the module. The way you're using it webpack will load the module, but the module then just registers some stuff in a global variable which you later access. This is kinda hacky and error-prone.

I don't know if it's easy to migrate here, but get rid of the `OCA.XXX = xxx` stuff. Don't use those globals and instead, have webpack locate the module and inside the module you export whatever you previously registered as global variable.

Does that make sense? :)

Reply:
See https://github.com/nextcloud/twofactor_u2f/pull/74/commits/ae4fe1c90a276013b3710a22c3aa1cdd6eb16db5#diff-ccd4d3da9150404f0f6690291ba3add9 as an example. The view class is exported because it's the returned value of the module definition.

Reply:
Worked fine for the components I think.
in app.js the exporting to OCA.Notifications is okay I guess?

Reply:
> in app.js the exporting to OCA.Notifications is okay I guess?

Is `OCA.Notifications` used elsewhere, outside of this app?

Reply:
If not, how about making `app.js` a module like the other components which you then load in a `init.js` script where you start the app like

```js
define(function(require) {
    'use strict';

    var $ = require('jquery')
    var App = require('./app')

    $(function() {
        var app = new App()
        app.initialize()
    })
})
```

Reply:
ReferenceError: define is not defined

😢

Reply:
is reported for app.js but I dont get why :/

Reply:
Hmmmm but you don't include the `app.js` file directly, right? It's bundled with webpack.

Reply:
Yeah, tested it a bit, when I remove `import Vue from 'vue';` it works. should I move that somewhere else, or replace it with a different call?